### PR TITLE
Update app-admin/adcli to 0.9

### PR DESCRIPTION
**Pull request by t-lo**
This PR updates `app-admin/adcli` to version 0.9.0 and updates the build recipe to use freedesktop.org's gitlab (since anongit and the package tarball server were retired).

Version 0.8.2 uses deprecated crypt algorithms and does not work in some scenarios with the new `mit-krb5` version 1.18.5 we added in 2605. This results in `adcli` 0.8.2 being unable to join AD; see https://github.com/flatcar-linux/Flatcar/issues/194 for details.

# Testing done

Built package and qemu image locally, successfully ran smoke test (`adcli --help`).

### Also cherry-pick for 2632 and main when merging.

### CODE REVIEW

File: app-admin/adcli/Manifest [hunk 1]
```diff
@@ -1 +1,2 @@
-DIST adcli-0.8.2.tar.gz 467168 SHA256 72f6db406e35d96de2bdc413a5ed69f28a4a735c08670c6556713c3f83921aa4 SHA512 a46e3f4b3c5434557a75cfe1c44c8bc7e9e7c7e240fa3a903e0095ef58505c2bcc66e80aa7b9a6bcf3284aed1d9af4068037c57cd5bd9f68a0bde34f429c44e9 WHIRLPOOL d7c8f8b028169dd29f5001762197ef9fd27e66df2711f2a7cd5ae9eed6d6c14c58778e968383b183eb62b0774a183c6031bb7b67f60fb9e4bd3bf3c029c4e6e1
+DIST adcli-0.8.2.tar.bz2 74600 SHA512 bc5624e7876b3bc9f481dc483b9606b127c953907c6553b5fe67c6c5ad9d081fe163037a985ba28f7e092baf12affc853938b6da4b58c1999a5173036c08c9b0 BLAKE2B cc6f5152a68c61e0fc4dc7f1607760cd5e5dd58ee4812416caa141e4cf4e2e9fe74351f193bda5f02045570bba5b2bf359fdc2c95f89fa985491c33f46543e02
```
Comment (dongsupark) [inline app-admin/adcli/Manifest:1]: we do not need the hash for 0.8.2.

**Review by dongsupark — APPROVED** [review]
Looks good.

File: app-admin/adcli/Manifest (hunk 1 above)
Comment (t-lo) [reply app-admin/adcli/Manifest:1]: Good point, removing.

**Comment by t-lo** [comment]
Merged and cherry-picked to [2632](https://github.com/flatcar-linux/coreos-overlay/commit/ff706f11e0b7dbee7100a4c1e66ff79fd88af487) and to [main](https://github.com/flatcar-linux/coreos-overlay/commit/7b1e8ce9fcdfa52bc4616f85ab2055da2d92e7b7).

**Pull request merged**